test/unit/metadata_test.py

```diff
@@ -39,3 +42,60 @@ def test_get_etl_group_tables() -> None:
     """Test that a Value error is raised for non existent etl group."""
     with pytest.raises(ValueError):
         Package.get_etl_group_tables("not_an_etl_group")
+
+
+@pytest.mark.parametrize("resource_name", sorted(PUDL_RESOURCES.keys()))
+def test_pyarrow_schemas(resource_name: str):
+    """Verify that we can produce pyarrow schemas for all defined Resources."""
+    _ = PUDL_RESOURCES[resource_name].to_pyarrow()
+
+
+@pytest.mark.parametrize("field_name", sorted(FIELD_METADATA.keys()))
+def test_field_definitions(field_name: str):
+    """Check that all defined fields are valid."""
+    _ = Field(name=field_name, **FIELD_METADATA[field_name])
+
+
+@pytest.mark.xfail(reason="Need to purge unused fields. See issue #3224")
+def test_defined_fields_are_used():
+    """Check that all fields which are defined are actually used."""
+    used_fields = set()
+    for resource in PUDL_RESOURCES.values():
+        used_fields |= {f.name for f in resource.schema.fields}
+    defined_fields = set(FIELD_METADATA.keys())
+    unused_fields = sorted(defined_fields - used_fields)
+    if len(unused_fields) > 0:
+        raise AssertionError(
+            f"Found {len(unused_fields)} unused fields: {unused_fields}"
+        )
+
+
+@pytest.mark.xfail(reason="Incomplete descriptions. See issue #3224")
+def test_fields_have_descriptions():
+    """Check that all fields have a description and report any that do not."""
+    fields_without_description = []
+    for field_name in FIELD_METADATA:
+        field = Field(name=field_name, **FIELD_METADATA[field_name])
+        if field.description is None:
+            fields_without_description.append(field_name)
+    fields_without_description = sorted(fields_without_description)
+    if len(fields_without_description) > 0:
+        raise AssertionError(
+            f"Found {len(fields_without_description)} fields without descriptions: "
+            f"{fields_without_description}"
+        )
+
+
+@pytest.mark.xfail(reason="Incomplete descriptions. See issue #3224")
+def test_resources_have_descriptions():
+    """Check that all resources have a description and report any that do not."""
+    resources_without_description = []
+    for resource_name, resource in PUDL_RESOURCES.items():
+        if resource.description is None:
+            resources_without_description.append(resource_name)
+    resources_without_description = sorted(resources_without_description)
+    if len(resources_without_description) > 0:
+        raise AssertionError(
+            f"Found {len(resources_without_description)} resources without descriptions: "
+            f"{resources_without_description}"
+        )
```

Comment:
Once #3224 has been finished and we are requiring all resources and fields to have descriptions we can remove these, but they were useful in compiling the lists of things that need to be fixed.

Reply:
I think these tests are worth keeping around after we complete #3224. We might add typos and extra fields into the system again.

Reply:
Those errors would be caught by Pydantic when it attempts to instantiate all of the Resources in the Package, and all of the Fields in the Resources, which it currently does (for better or worse) on import, if we make `description` a required field, and extra fields are already disallowed.

Reply:
We just can't enforce that now since we have 15 resources with no description and 150 fields with no description, and 42 fields that we aren't using so they never get validated by Pydantic.

Reply:
Oh I see. That makes sense 👍